dockerfiles/substrate-ci-linux/Dockerfile

```diff
@@ -18,23 +18,25 @@ dockerfiles/rust-builder/README.md" \
 
 WORKDIR /builds
 
-# Mozilla's Ubuntu PPA key
-COPY utility/mozilla.key /etc/apt/trusted.gpg.d/ubuntu-mozilla.gpg
-
 # install tools and dependencies
 RUN set -eux; \
-	echo "deb http://ppa.launchpad.net/ubuntu-mozilla-security/ppa/ubuntu bionic main" \
-		> /etc/apt/sources.list.d/mozilla.list; \
 	apt-get -y update; \
 	apt-get install -y --no-install-recommends \
-		firefox; \
+		firefox-esr; \
+# install geckodriver
+	curl -s https://api.github.com/repos/mozilla/geckodriver/releases/latest \
+		| grep 'browser_download_url.*linux64.tar.gz"' \
+		| cut -d : -f 2,3 | tr -d \" | xargs -n 1 curl -sSL \
+		| tar -xzC /usr/bin; \
```

Comment:
these kind of files should go to `/usr/loca/bin` in order not to override existing files. also `xargs -n 1` runs curl with one file at a time so should there be multiple files to download they will all be downloaded and unpacked to that folder. the easy way out would be to pipe it to `tail -n1` and use `sort -V` before. would it make sense to fix it to one version for reproducibilty reasons?